### PR TITLE
Make some minor cleanups to aggregators

### PR DESCRIPTION
**Pull request by ronawho**
Minor aggregator cleanup in prep for some larger refactoring.

### CODE REVIEW

File: src/SegmentedArray.chpl [hunk 1]
```diff
@@ -196,7 +196,7 @@ module SegmentedArray {
       // NOTE: cannot compute lengths inside forall because agg.copy will
       // experience race condition with loop-private variable
       var right: [D] int, left: [D] int;
-      forall (r, l, idx) in zip(right, left, iv) with (var agg = new SrcAggregator(int)) {
+      forall (r, l, idx) in zip(right, left, iv) with (var agg = newSrcAggregator(int)) {
```
Comment (mhmerrill) [inline src/SegmentedArray.chpl:199]: did you mean to remove the space? where is the proc newSrcAggregator?

Comment (ronawho) [reply src/SegmentedArray.chpl:199]: Yeah meant to remove. `newSrcAggregator` and `newDstAggregator` are the wrappers that select between aggregation and unordered copy (based off comm={none,ugni,etc.})

Previously, this was always forcing aggregation even though that's not what you want for comm=none (and currently for ugni.) Everything else uses the wrappers, this was the one spot that didn't.

Comment (mhmerrill) [reply src/SegmentedArray.chpl:199]: Ok, sounds good.

**Review by mhmerrill — APPROVED** [review]
Looks good.

**Pull request merged**